shell/platform/android/io/flutter/embedding/android/FlutterView.java

```diff
@@ -1449,6 +1449,13 @@ public void removeFlutterEngineAttachmentListener(
         .send();
   }
 
+  private FlutterViewDelegate delegate = new FlutterViewDelegate();
+
+  @VisibleForTesting
+  public void setDelegate(FlutterViewDelegate delegate) {
+    this.delegate = delegate;
+  }
+
   private void sendViewportMetricsToFlutter() {
```

Comment:
Consider adding a link to https://github.com/flutter/engine/blob/main/lib/ui/hooks.dart#L139-L155 in the ViewportMetrics class (not here) so that future callers can more easily find how it is used.